ynput/OpenPype
# General: Push to studio library

## Brief description
Added simple action to push version to a library project.

## Description
Added action "Push to library project" which can run only on one version at a time. The action is triggering new tool "push to project" which fake's integration. This is first iteration over the feature as prove of concept which means "it is not perfect". The premise is to be able push version (and asset) to library project with representations that are integrated as would be during publishing. The tool gives user option to select context to which is version pushed, what variant will be used for subset name and ability to create new asset. This can cause issues as the creation of asset is not propagated to project managers (ftrack, shotgrid, kitsu, ...). Also there is missing review or thumbnail.

Added `LoadError` to show custom message to a user when loading fails.

## Additional info
~~- User can trigger push to different project only once and any crash will cause that he must trigger the action again. At this moment it is because the overlay is showing "result" of the process and give option to only close the window, maybe "Try again" button could be added -> showed only when process fails?~~ Added `"Try again"` button if process fails with known error.
- Maybe add settings to be able disable the action -> not sure if all studios would like to use it?

### Unrelated info
- Discovered small issue in loader, I've found out is that loaders can't define full name of icon in font awesome library, only icons starting with `fa.`.

## Screenshot
![image](https://user-images.githubusercontent.com/43494761/211001038-5697905f-e528-4a7a-97be-c694e4371c9d.png)


## Testing notes:
1. Open Loader/Library loader
2. Select an asset which has published versions
3. Select a version with representations
4. Right click on the version a hit `"Push to Library project"`
5. New process should start with UI tool
6. Select library project where to publish
7. Select asset
*8. Select task (only if `"New asset name"` is empty)
9. Fill/Remove `"New asset name"` value
    - If is filled new asset is created and selected asset is used as parent
    - Selected asset is used as target asset if not filled
10. Check/fill variant for subset name
*11. Add comment
12. Hit `Publish`
13. Wait until process finishes
